### PR TITLE
Document new prepareForValidation method

### PR DESCRIPTION
**Pull request by jtgrimes**
The wording isn't perfect, but it's something.

### CODE REVIEW

File: validation.md [hunk 1]
```diff
@@ -261,6 +262,22 @@ If you plan to have authorization logic in another part of your application, sim
         return true;
     }
 
+<a name="modifying-data-before-validation"></a>
```
Comment (rscafi) [inline validation.md]: I guess that it should be `<a name="executing-data-before-validation"></a>` so that the anchor works

Comment (mdavis1982) [inline validation.md]: Shouldn't it be `executing-code-before-validation` as in the link above?

Comment (jtgrimes) [inline validation.md]: Thanks. Will update this afternoon.

**Comment by taylorotwell** [comment]
Can you send this to master branch?

**Pull request closed without merging**